[#16437] fix sync qr code not recognized after trying again

ulisesmac opened the pull request:
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #16437

### Summary

This PR solves the problem of the QR code not being recognized after the attempt failed and the user pressed the "try again" button.


### Review notes
This PR has important refactoring changes and the solution involves subscribing this component to the lifecycle react methods provided by react native navigation:
https://wix.github.io/react-native-navigation/7.25.4/api/events/#registercommandlistener

I added a helper function, that I'd like it to be a macro to make things easier, but I'm not sure, since adding a macro could make developer lifes harder sometimes, so I want to get your opinion. Please read the comments in the code of this PR.

#### Platforms
<!-- (Optional. Specify which platforms should be tested) -->

- Android
- iOS

#### Areas that maybe impacted

- Onboarding
- Create a new account

### Steps to test
<!-- (Specify exact steps to test if there are such) -->

Generate a valid but expired QR code for syncing, you can use this:
![image](https://github.com/status-im/status-mobile/assets/90291778/168bb7e9-bbc5-4139-9c79-41347ebfef85)


- Open Status in a fresh install
- Click  Sign in button
- Scan the code using the phone/emulator
- An error will be shown
- Click try again button
- You are now able to rescan the code again.




status: ready

ulisesmac committed:
Add bind-component function to `react-native.navigation`

ulisesmac committed:
Create helper function to use react-native-navigation lifecycle methods

ulisesmac committed:
Fix Sync QR Code not recognized after trying again & refactor

ulisesmac commented on src/react_native/navigation.cljs:
```diff
@@ -89,3 +89,8 @@
                             :bottom-tabs-height (.-bottomTabsHeight consts)
                             :status-bar-height  (.-statusBarHeight consts)})))
+
+(defn bind-component
+  [^js/Object this component-id]
+  (set! (. this -navigationEventListener)
+    (.. Navigation events (bindComponent this component-id))))
```
This function allow us to create a react class and use the react-native-navigation lifecycle methods to detect when a component has appeared or disappeared from user's screen

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/animation.cljs:
```diff
@@ -0,0 +1,61 @@
+(ns status-im2.contexts.syncing.scan-sync-code.animation
```
Just moved the animation code from `view` to this namespace, since it was hard to read ther. I hope now it's clearer

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -48,8 +50,8 @@
   (rf/dispatch [:syncing/preflight-outbound-check #(reset! preflight-check-passed? %)]))
 
-(defn- f-header
-  [{:keys [active-tab read-qr-once? title title-opacity subtitle-opacity reset-animations-fn animated?]}]
```
`read-qr-once?` was an atom to control the callbacks dispatched by the camera-kit component, it dispatched the `on-read` function many times before we navigated.

Now it's not required

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -119,12 +121,7 @@
-                          (reset! active-tab id)
-                          (reset! read-qr-once? false))}]]]))
-
-(defn- header
-  [props]
-  [:f> f-header props])
```
We don't need this wrapper, we use `:f>` wrappers to expose an API, for internal component use, we can directly use  the `:f>` call

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 8e010d10 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-16746/2/) | 2023-07-22 01:47:14 | ~5 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-230722-014128-8e010d-pr16746-universal.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-230722-014128-8e010d-pr16746-universal.apk)|

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -255,136 +252,102 @@
-                                       connection-string]
-                                      300)
+      (do
+        (set-qr-code-succeeded)
```
`set-qr-code-succeeded` is a function that will change a control atom, we need it to not scanning the qr code more than once.

The behaviour is similar to the  previous `read-qr-once?`  atom, but now it is handled by the `compnentDidAppear` method

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -255,136 +252,102 @@
-        :overlay-color    colors/neutral-80-opa-80
-        :background-color colors/neutral-80-opa-80}]]]))
+(defn- render-camera
+  [{:keys [torch-mode qr-view-finder set-qr-code-succeeded]}]
+  [:<>
+   [rn/view {:style style/camera-container}
+    [camera-kit/camera
+     {:style        style/camera-style
+      :camera-type  camera-kit/camera-type-back
+      :zoom-mode    :off
+      :torch-mode   torch-mode
+      :scan-barcode true
+      :on-read-code #(check-qr-code-and-navigate % set-qr-code-succeeded)}]]
+   [hole-view/hole-view
+    {:style style/hole
+     :holes [(merge qr-view-finder {:borderRadius 16})]}
+    [blur/view
+     {:style            style/absolute-fill
+      :blur-amount      10
+      :blur-type        :transparent
+      :overlay-color    colors/neutral-80-opa-80
+      :background-color colors/neutral-80-opa-80}]]])
```
Small refactor:

This `render-camera` component has a `when` and it could return `nil`. Instead of creating a component that is returning nil, now `render-camera` always renders the camera, and the condition to show it or not is when calling it.

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -255,136 +252,102 @@
-        qr-view-finder     (reagent/atom {})
-        render-camera?     (reagent/atom false)
-        torch?             (reagent/atom false)
-        app-state-listener (atom nil)]
```
We don't need this `app-state-listener` atom, explanation below

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -255,136 +252,102 @@
+            (when @dismiss-animations (@dismiss-animations))
+            (animation/reset-animations params))]
     (fn []
-      (let [camera-ref (atom nil)
-            read-qr-once? (atom false)
```
These atoms shouldn't be declared in the render function since they will be recreated in each render

smohamedjavid replied:
@ulisesmac - The `on-read-code` function from the `react-native-camera-kit` library is called as long as a QR code is in the frame of the camera (or the user is focusing on it). To prevent this, we added the `read-qr-once?` atom within the render and reset it after 3 seconds on a successful read.

This also allows the user to scan any QR code after 3 seconds if they accidentally scanned the wrong one. 

The present code will result in endless error toasts if the user scans a wrong PR.

J-Son89 replied:
Hmm, is there any way to add a unit test (component spec fine) to ensure this feature is retained?

ulisesmac replied:
@smohamedjavid 
The present code is using a different approach without using the atom in the render function and also fixing the bug for this issue.

What would be a wrong QR?
Do you know if that endless toast behavior is reproducible in this branch? Could you please tell me how to reproduce it?

Thank you so much!

smohamedjavid replied:
@ulisesmac 

> What would be a wrong QR?

A QR code which is NOT a sync QR (QR data would start with `cv2:`). 

> Do you know if that endless toast behavior is reproducible in this branch? Could you please tell me how to reproduce it?

Yes, Kindly download and install this PR build (https://github.com/status-im/status-mobile/pull/16746#issuecomment-1646374593) into your device. Point your device at any QR code from the internet.

It doesn't need to be a wrong QR. The camera library fires the `on-read-code` method (with QR data) as long as a QR is in focus. This will happen with the correct sync QR too (initiates the local pairing repeatedly by calling the native module as long as a Correct Sync QR is in focus).

<hr />

I fixed a similar issue https://github.com/status-im/status-mobile/issues/15970 (PR: https://github.com/status-im/status-mobile/pull/16012) with the help of `view-id` in `app-db`.

smohamedjavid replied:
> Hmm, is there any way to add a unit test (component spec fine) to ensure this feature is retained?

I don't know as this is a ratom inside render (not a prop). But, I will see if we can add a check. 

P.S. I would add a comment explaining why we need it.

ulisesmac replied:
@smohamedjavid 

The problem I see with a non-reactive atom within the render function is that we are not able to know what's the value stored by the atom: Let's say we reset the atom, then we `deref` it, if a re-render happens just before, our value will be missed.

Thanks for noticing this case, I didn't test with a wrong QR (so bad).

I added a reagent  control atom to avoid multiple calls to `on-read` (not inside the render function), also the toast call is using `debounce-and-dispatch` in the same way that when we are navigating. :+1: 

The commit where I added is 2279613f7f249bc90eecc3c8c8d90e00bb829358,
please review again and approve if this is ok for you, otherwise we can check what else is missing :)

Thanks!

smohamedjavid replied:
This seems to be working fine. 👍 

Thanks, @ulisesmac!

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -255,136 +252,102 @@
-        render-camera?     (reagent/atom false)
-        torch?             (reagent/atom false)
-        app-state-listener (atom nil)]
+(defn- reset-animations-and-navigate-fn
+  [{:keys [render-camera? show-camera?] :as params}]
+  (letfn [(reset-fn []
+            (rf/dispatch [:navigate-back])
+            (when @dismiss-animations (@dismiss-animations))
+            (animation/reset-animations params))]
```
Refactor: just extracted the animations to a separate function to make the reading of the main view function easier

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -255,136 +252,102 @@
+      (reset! render-camera? false)
+      (js/setTimeout reset-fn (if show-camera? 500 0)))))
+
+(defn- set-listener-torch-off-on-app-inactive
+  [torch-atm]
+  (let [set-torch-off-fn   #(when (not= % "active") (reset! torch-atm false))
+        app-state-listener (.addEventListener rn/app-state "change" set-torch-off-fn)]
+    #(.remove app-state-listener)))
```
Again a refactor. just extracted this effect callback to a separate function

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -255,136 +252,102 @@
-                                                              0
-                                                              :easing4))
-               (if show-camera? 500 0)))]
-        (rn/use-effect (fn []
-                         (reset! app-state-listener
-                           (.addEventListener rn/app-state
-                                              "change"
-                                              #(when (and (not= % "active") @torch?)
-                                                 (reset! torch? false))))
-                         #(.remove @app-state-listener)))
```
About the `app-state-listener` atom removal:

it's not required to be an atom, just a `let` binding:

```clojure
(let [res (.addEventListener something "something else" fn-callback)]
  #(.remove res))
```

ulisesmac commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -434,5 +403,14 @@
 (defn view
-  [props]
-  [:f> f-view props])
+  [_props]
+  (let [qr-code-succeed? (reagent/atom false)]
+    (navigation.util/create-class-and-bind
+     "sign-in-intro"
+     {:component-did-appear (fn set-qr-code-failed [_this]
+                              (reset! qr-code-succeed? false))}
+     (fn [props]
+       (let [new-pops (assoc props
+                             :qr-code-succeed?      @qr-code-succeed?
+                             :set-qr-code-succeeded #(reset! qr-code-succeed? true))]
+         [:f> f-view new-pops])))))
```
This is the important part of the solution:

I am using the new helper function I mentioned in this PR description. as I said before, I'd like it to be a macro.

It will reset the atom `qr-code-suceeed?` to `false` each time this screen is shown to the user, so we can read the code again.

In react-native-navigation, the screen components are not unmounted when we navigate to another screen within the same stack, it means:
- the Cleanup function in `use-effect`
- the `finally` statement in `with-let`
- the componentWillUnmount() method

will not be called until the screen is detached from navigation hierarchy, but in cases like this (where we want to go back and we expect the screen state is reset) we need to use the `componentDidAppear` method given by `react-native-navigation`.

Now that this has been noticed, I think we will start using these extended lifecycle methods more often, so I decided to create a helper function.

ulisesmac commented on src/status_im2/navigation/util.cljs:
```diff
@@ -0,0 +1,42 @@
+(ns status-im2.navigation.util
```
I didn't know where to add this new function without adding dependency problems, so I created this namespace.

Please take a look at the function and give any feedback you think will enhance it.

Do you think a macro would be good?
we could avoid repetition and I like it, but at the same time, I don't like existing macros like `defview` that we had in the previous codebase since they add a layer of complexity to our components.
wdyt? :thinking:

smohamedjavid commented on src/status_im2/contexts/syncing/scan_sync_code/view.cljs:
```diff
@@ -434,5 +403,14 @@
+  [_props]
+  (let [qr-code-succeed? (reagent/atom false)]
+    (navigation.util/create-class-and-bind
+     "sign-in-intro"
```
I believe the name should be received in a prop because we use this view (as the screen) in two places:

1. Sign In
2. Scan sync code

smohamedjavid replied:
Should we match it with the name declared in `screens.cljs`?
https://github.com/status-im/status-mobile/blob/eb3de978cef417673fabe21eb1c60fc72f905e27/src/status_im2/navigation/screens.cljs#L210

ulisesmac replied:
Done, the `screen-name` is received as a parameter

ulisesmac committed:
Add screen name as parameter

ulisesmac committed:
Add atom to manage callbacks when qr code scan fails

status-im-auto commented:
## 85% of end-end tests have passed
```
Total executed tests: 39
Failed tests: 6
Passed tests: 33
Not executed tests: 1
```
```
IDs of not executed tests: 702869 
```
```
IDs of failed tests: 702732,703133,702851,702894,702731,703382 
```
<h3>Not executed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=16746&TR_CASE_IDS=702869">Rerun not executed tests</a></li></details><h3>Failed tests (6)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=16746&TR_CASE_IDS=702732,703133,702851,702894,702731,703382">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202209">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Looking for a message by text: Hurray! unblocked </div></blockquote></p>

```
critical/test_public_chat_browsing.py:905: in test_community_contact_block_unblock_offline
    self.errors.verify_no_errors()
base_test_case.py:183: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Hurray! unblocked was not received in public chat after user unblock!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#366">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#349">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202215">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `show-profiles` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/test_public_chat_browsing.py:486: in test_restore_multiaccount_with_waku_backup_remove_switch
    self.errors.verify_no_errors()
base_test_case.py:183: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 admin_open was not restored from waku-backup!!
E    member_open was not restored from waku-backup!!
E    admin_closed was not restored from waku-backup!!
E    member_closed was not restored from waku-backup!!
E    Incorrect contacts number restored: 0 instead of 2
E    Contact(s) was (were) not restored from backup: Test_contact, MyCustomNickname!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f86ce720e7b4fab87a40ec06e05b6f3?auth=02b93a759914931f0ea25bb300599751#207">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202219">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `MuteButton` by `accessibility id`: `mute-community` </div><div>Device 1: Click system back button </div></blockquote></p>

```
critical/test_public_chat_browsing.py:402: in test_community_mute_community_and_channel
    self.errors.verify_no_errors()
base_test_case.py:183: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Text 'Muted until 08:52 Thu 03 Aug' is not shown for a muted community channel
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f86ce720e7b4fab87a40ec06e05b6f3?auth=02b93a759914931f0ea25bb300599751#118">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202185">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Pin feature is in development]]</b><br/><br/></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202184">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Pin feature is in development]]</b><br/><br/></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202205">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Button` by `accessibility id`: `tab-contacts` </div><div>Device 2: Tap on found: Button </div></blockquote></p>

```
medium/test_activity_center.py:112: in test_activity_center_contact_request_accept_swipe_mark_all_as_read
    self.errors.verify_no_errors()
base_test_case.py:183: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Contact was not added to contact list after accepting contact request (as receiver)
E    Contact was not added to contact list after accepting contact request (as sender)
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7c118b97e8e345298f2552d5c48b02a8?auth=253f467109d57baeb08292ec6a028543#90">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fea749b609c84b14a3dd6916b16c080d?auth=cb3e33bec6b07d33bac15e8fe2a8681d#93">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (33)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202216">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#205">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#155">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202207">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#230">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#213">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202198">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#279">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#281">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202214">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#498">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#372">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202192">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#518">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#396">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202197">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#190">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#126">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202196">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#148">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#97">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202201">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#300">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#296">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202200">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#166">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#114">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202202">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#533">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#419">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202199">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/de93c63050c44ef7b5bb6cef582f56d9?auth=09d2d0526f09bab2e0a0b5a5115a7c22#341">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44600dae6b1c4cffb5f4229069558351?auth=4f0495ba973c307c273ab781b83d80d2#339">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202187">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f86ce720e7b4fab87a40ec06e05b6f3?auth=02b93a759914931f0ea25bb300599751#93">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202203">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9f86ce720e7b4fab87a40ec06e05b6f3?auth=02b93a759914931f0ea25bb300599751#75">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202221">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a1d12d7a9dfe489fb6997e41bc576513?auth=7b5f65258fd8dc85698493e910300aed#330">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44a8ac9f2cc3427aa967133a43f049d2?auth=6630aaa2f2a61e45d24abfb050dc81ce#214">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/82acc50c3c4d461599b71fcc8f05dbcd?auth=2f39d68b5ec815bab98be5d48d48cb27#202">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202218">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a1d12d7a9dfe489fb6997e41bc576513?auth=7b5f65258fd8dc85698493e910300aed#240">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44a8ac9f2cc3427aa967133a43f049d2?auth=6630aaa2f2a61e45d24abfb050dc81ce#156">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/82acc50c3c4d461599b71fcc8f05dbcd?auth=2f39d68b5ec815bab98be5d48d48cb27#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202217">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a1d12d7a9dfe489fb6997e41bc576513?auth=7b5f65258fd8dc85698493e910300aed#141">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44a8ac9f2cc3427aa967133a43f049d2?auth=6630aaa2f2a61e45d24abfb050dc81ce#98">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/82acc50c3c4d461599b71fcc8f05dbcd?auth=2f39d68b5ec815bab98be5d48d48cb27#78">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202193">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a1d12d7a9dfe489fb6997e41bc576513?auth=7b5f65258fd8dc85698493e910300aed#125">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44a8ac9f2cc3427aa967133a43f049d2?auth=6630aaa2f2a61e45d24abfb050dc81ce#71">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/82acc50c3c4d461599b71fcc8f05dbcd?auth=2f39d68b5ec815bab98be5d48d48cb27#71">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202194">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a1d12d7a9dfe489fb6997e41bc576513?auth=7b5f65258fd8dc85698493e910300aed#277">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/44a8ac9f2cc3427aa967133a43f049d2?auth=6630aaa2f2a61e45d24abfb050dc81ce#189">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/82acc50c3c4d461599b71fcc8f05dbcd?auth=2f39d68b5ec815bab98be5d48d48cb27#177">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202189">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#128">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#108">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202186">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#392">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#335">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202191">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#539">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#468">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202195">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#256">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#195">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202188">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#190">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#149">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202190">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#445">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#402">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202222">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#479">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#455">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202206">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#303">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#224">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202220">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#328">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#264">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202183">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/24a6d8be0090496585b4f5f05e28eaf6?auth=e9faddbb29734d442e953fd9287e7ad8#78">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e03cfd7a40324dd48de183d7e5d9d5c3?auth=2d60282d33f39123f56eb45ddb51d615#77">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202212">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/89407a6a961940c0a753a8285c5ee319?auth=c88f6ba8edf55a109ebbf99fd8fde397#277">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d7db92c2189d4558afe57f6dd63b3691?auth=c5854d65e1212b1af3e2e1f0ea34b6bc#126">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202210">test_navigation_jump_to</a>, id: 702936 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/89407a6a961940c0a753a8285c5ee319?auth=c88f6ba8edf55a109ebbf99fd8fde397#144">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d7db92c2189d4558afe57f6dd63b3691?auth=c5854d65e1212b1af3e2e1f0ea34b6bc#107">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202211">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/89407a6a961940c0a753a8285c5ee319?auth=c88f6ba8edf55a109ebbf99fd8fde397#165">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d7db92c2189d4558afe57f6dd63b3691?auth=c5854d65e1212b1af3e2e1f0ea34b6bc#110">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202213">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/89407a6a961940c0a753a8285c5ee319?auth=c88f6ba8edf55a109ebbf99fd8fde397#313">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d7db92c2189d4558afe57f6dd63b3691?auth=c5854d65e1212b1af3e2e1f0ea34b6bc#144">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3202204">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7c118b97e8e345298f2552d5c48b02a8?auth=253f467109d57baeb08292ec6a028543#55">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fea749b609c84b14a3dd6916b16c080d?auth=cb3e33bec6b07d33bac15e8fe2a8681d#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

churik commented:
@ulisesmac 
after scanning invalid QR > try again I was redirected to success screen (both IOS and Android)

Example of QR:
<img width="137" alt="invalid_qr" src="https://github.com/status-im/status-mobile/assets/4557972/caf865ac-1d4a-4f18-9a1e-03d1474a4344">


Actual result: 

https://github.com/status-im/status-mobile/assets/4557972/2b3f09c3-de73-4a41-884e-62c38c7097e6

ulisesmac commented:
> @ulisesmac after scanning invalid QR > try again I was redirected to success screen (both IOS and Android)
> 
> Example of QR: <img alt="invalid_qr" width="137" src="https://user-images.githubusercontent.com/4557972/256822994-caf865ac-1d4a-4f18-9a1e-03d1474a4344.png">
> 
> Actual result:
> FILE.2023-07-28.14.38.21.mp4

Hi @churik 
Could you please share the QR used to test? I tried with the given QR and I couldn't replicate the issue.

Also, is this an issue present in develop or introduced in this PR? since this PR is not touching the code of clicking that button.

Thanks!

churik commented:
Ok, may be I misunderstood something, but I followed the steps mentioned in https://github.com/status-im/status-mobile/issues/16437

I used QR mentioned https://github.com/status-im/status-mobile/pull/16746#issuecomment-1655623200 and I got the same behavior in nightly and in this PR (so Sign In > Scan QR > Try again > Syncing - nothing happens, I cannot scan QR again)
![image](https://github.com/status-im/status-mobile/assets/4557972/1753028e-f249-4bbb-88d0-8261333c6388)
Reproducible **only on Android.**

ulisesmac committed:
Fix screen name

ulisesmac commented:
> Ok, may be I misunderstood something, but I followed the steps mentioned in #16437
> 
> I used QR mentioned [#16746 (comment)](https://github.com/status-im/status-mobile/pull/16746#issuecomment-1655623200) and I got the same behavior in nightly and in this PR (so Sign In > Scan QR > Try again > Syncing - nothing happens, I cannot scan QR again) ![image](https://user-images.githubusercontent.com/4557972/257157263-1753028e-f249-4bbb-88d0-8261333c6388.png) Reproducible **only on Android.** (apologies for not mentioning it earlier)

Hello again @churik 

I tested this flow on develop. I selected  the option "Add existing Status profile" on the "Profiles on device" screen and I replicated the issue you mentioned. So it seems this flow is broken on develop, that's not related to this PR, so let's create an issue if it doesn't exist yet.

To test this PR, you can open Status in a fresh install and select the "sign in" option. I updated the PR's description and added a QR to test, but it can also be replicated with the QR you added.

Could you please test it again? thanks!

churik commented:
@ulisesmac 
Now in this PR:

1) Sign in
2) Scan QR provided in the description here: https://github.com/status-im/status-mobile/pull/16746#issue-1816530045 - nothing happens 
https://github.com/status-im/status-mobile/assets/4557972/a23241ac-d891-4e21-8214-3b5f60815723

Logs: 
[Status-debug-logs (1).zip](https://github.com/status-im/status-mobile/files/12260355/Status-debug-logs.1.zip)

Also from the video here: https://github.com/status-im/status-mobile/pull/16746#issuecomment-1655623200 I can see that I repeated exact steps from description: https://github.com/status-im/status-mobile/pull/16746#issue-1816530045, so frankly still not sure what I was doing wrong here in order to reproduce the issue

ulisesmac commented:
> @ulisesmac 
> 
> Now in this PR:
> 
> 
> 
> 1) Sign in
> 
> 2) Scan QR provided in the description here: https://github.com/status-im/status-mobile/pull/16746#issue-1816530045 - Nothing happens.
> 
> 
> 
> https://github.com/status-im/status-mobile/assets/4557972/a23241ac-d891-4e21-8214-3b5f60815723
> 
> 
> 
> Logs: 
> 
> [Status-debug-logs (1).zip](https://github.com/status-im/status-mobile/files/12260355/Status-debug-logs.1.zip)
> 
> 
> 
> Also from the video here: https://github.com/status-im/status-mobile/pull/16746#issuecomment-1655623200 I can see that I repeated exact steps from description: https://github.com/status-im/status-mobile/pull/16746#issue-1816530045, so frankly still not sure what I was doing wrong here in order to reproduce the issue
> 
> 

Hmmm... 🤔 I will check, thanks!

ulisesmac commented:
> > @ulisesmac
> > Now in this PR:
> > 
> > 1. Sign in
> > 2. Scan QR provided in the description here: [[#16437] fix sync qr code not recognized after trying again #16746 (comment)](https://github.com/status-im/status-mobile/pull/16746#issue-1816530045) - Nothing happens.
> > 
> > 
> >   
> >     
> >     
> > 
> >     FILE.2023-08-04.13.49.30.mp4
> >     
> >   
> > 
> >   
> > 
> >   
> > 
> > 
> > Logs:
> > [Status-debug-logs (1).zip](https://github.com/status-im/status-mobile/files/12260355/Status-debug-logs.1.zip)
> > Also from the video here: [#16746 (comment)](https://github.com/status-im/status-mobile/pull/16746#issuecomment-1655623200) I can see that I repeated exact steps from description: [#16746 (comment)](https://github.com/status-im/status-mobile/pull/16746#issue-1816530045), so frankly still not sure what I was doing wrong here in order to reproduce the issue
> 
> Hmmm... thinking I will check, thanks!

@churik  A quick update on this.
The screen is not stuck, something is actually happening, but it takes too long (I guess is something related to a network timeout).
This issue is also happening on `develop`, the only one difference is that on develop the camera remains active and on this branch, I'm turning off the camera once the code has been recognized (to avoid multiple code recognitions).
I'm reproducing the issue you mention in develop in the following video:

https://github.com/status-im/status-mobile/assets/90291778/bbf80a98-27dc-4514-a5fc-57b84a571d19



I think it's still possible to test this PR (maybe with another QR, or using the one you posted as example), and we can report these problems as separate issues. It seems this flow has to be carefully reworked, but ti wasn't the original scope of this issue/PR.

ulisesmac committed:
Small refactor
